ts/test/session/sending/PendingMessageCache_test.ts

```diff
@@ -0,0 +1,216 @@
+// tslint:disable-next-line: no-require-imports no-var-requires
+const Data = require('../../../../js/modules/data');
+
+import { expect, assert } from 'chai';
+import sinon from 'sinon';
+import uuid from 'uuid';
+
+import { ChatMessage } from '../../../session/messages/outgoing';
+import { MessageUtils, PubKey } from '../../../session/utils';
+import { PendingMessageCache } from '../../../session/sending/PendingMessageCache';
+import { RawMessage } from '../../../session/types/RawMessage';
+import { SignalService } from '../../../protobuf';
+
+describe('PendingMessageCache', () => {
+  const sandbox = sinon.createSandbox();
+  let pendingMessageCacheStub: PendingMessageCache;
+
+  // tslint:disable-next-line: promise-function-async
+  const wrapInPromise = (value: any) =>
+    new Promise(r => {
+      r(value);
+    });
+
+  const generateUniqueMessage = (): ChatMessage => {
+    return new ChatMessage({
+      body: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+      identifier: uuid(),
+      timestamp: Date.now(),
+      attachments: undefined,
+      quote: undefined,
+      expireTimer: undefined,
+      lokiProfile: undefined,
+      preview: undefined,
+    });
+  };
+
+  beforeEach(async () => {
+    const mockStorageObject = wrapInPromise([] as Array<RawMessage>);
+    const voidPromise = wrapInPromise(undefined);
+
+    // Stub out methods which touch the database.
+    sandbox
+      .stub(PendingMessageCache.prototype, 'getFromStorage')
+      .returns(mockStorageObject);
+    sandbox
+      .stub(PendingMessageCache.prototype, 'syncCacheWithDB')
+      .returns(voidPromise);
+
+    // Initialize new stubbed cache
+    pendingMessageCacheStub = new PendingMessageCache();
+    await pendingMessageCacheStub.init();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('can initialize cache', async () => {
+    const { cache } = pendingMessageCacheStub;
```

Comment:
it would be better to have a test which creates a new instance than use one that is stored in a variable and set in `beforeEach`.

Also it'd be best if `cache` is made `private` and instead have a function `getCount` which returns the cache count or a `getCache` function which returns a copy of the cache.

The reason we should do this is because we don't want someone going in and modifying the internal cache of the object.